netpoll/handle.go

```diff
@@ -16,20 +16,31 @@ type filer interface {
 type Desc struct {
 	file  *os.File
 	event Event
+	fd    int
 }
 
 // NewDesc creates descriptor from custom fd.
 func NewDesc(fd uintptr, ev Event) *Desc {
-	return &Desc{os.NewFile(fd, ""), ev}
+	return &Desc{file: os.NewFile(fd, ""), event: ev, fd: int(fd)}
+}
+
+// NewDescFile creates descriptor from os.File
+func NewDescFile(file *os.File, ev Event) *Desc {
+	return &Desc{file: file, event: ev, fd: -1}
 }
 
 // Close closes underlying file.
 func (h *Desc) Close() error {
 	return h.file.Close()
 }
 
-func (h *Desc) fd() int {
-	return int(h.file.Fd())
+// Fd returns the underlying file descriptor
+func (h *Desc) Fd() int {
```

Comment:
Why this now must be exported?

Reply:
since I need the underlying fd in my scene so I have to export the fd:(.

Reply:
In my scene, I need the fd and join the fd to another epoll fd which is super strange but it works. I use easygo to solve https://github.com/golang/go/issues/15735